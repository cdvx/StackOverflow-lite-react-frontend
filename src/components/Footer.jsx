import React from 'react';

export default () => (
  <div>
    {/* <!-- Footer --> */}
    <footer className="page-footer relative-bottom font-small unique-color-dark">
      {/* <!-- Footer Elements --> */}
      <hr id="thisLine"/>
      <div className="container">
        {/* <!-- Grid row--> */}

        <div className="row">
          {/* <!-- Grid column --> */}
          <div className="col-md-12 py-5">
            <div className="mb-5 flex-center">
              {/* <!-- Facebook --> */}
              <a className="fb-ic">
                <i className="fa fa-facebook fa-lg white-text mr-md-5 mr-3 fa-2x" />
              </a>
              {/* <!-- Twitter --> */}
              <a className="tw-ic">
                <i className="fa fa-twitter fa-lg white-text mr-md-5 mr-3 fa-2x" />
              </a>
              {/* <!-- Google +--> */}
              <a className="gplus-ic">
                <i className="fa fa-google-plus fa-lg white-text mr-md-5 mr-3 fa-2x" />
              </a>
            </div>
          </div>
          {/* <!-- Grid column --> */}
        </div>
        {/* <!-- Grid row--> */}
      </div>
    <div className="footer-copyright text-center py-3">
      Â© 2018 Copyright:
        <a href="#">StackOverflow-lite.com</a>
      </div>
      {/* <!-- Copyright --> */}
    </footer>
  </div>
);
