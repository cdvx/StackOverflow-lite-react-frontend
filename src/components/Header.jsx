import React, { Component } from 'react';

class Header extends Component {

    logOut = event => {
        localStorage.clear();
    };

    render() {
        return (  
            <header>
                <nav  id="head-footer" className="navbar navbar-default">
                    <div className="container-fluid">
                        <div className="navbar-header">
                        <button type="button" className="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
                            <span className="sr-only">Toggle navigation</span>
                            <span className="icon-bar"></span>
                            <span className="icon-bar"></span>
                            <span className="icon-bar"></span>
                        </button>
                        <a className="navbar-brand " id="select-question" href="/"><p>StackOverflow-lite</p></a>
                        </div>
    
                        <div className="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
                        <ul className="nav navbar-nav">
                            <li className="active">{localStorage.getItem("token") && <a id="select-question" href="/postquestion">Post Question <span className="sr-only">(current)</span></a>}</li>
                            
                        </ul>
                        <ul className="nav navbar-nav navbar-right">
                            {!localStorage.getItem("token") && <li className="active"><a id="select-question" href="/login">Login</a></li>}
                            {!localStorage.getItem("token") && <li className="active"><a id="select-question" href="/signup">SignUp</a></li>}
                            {localStorage.getItem("token") && <li className="active"><a id="select-question" className="logOut" href="/" onClick={this.logOut} >Sign out</a></li>}
                        </ul>
                        </div>
                    </div>
                </nav>
            </header>
        );
    }
}

 
export default Header;