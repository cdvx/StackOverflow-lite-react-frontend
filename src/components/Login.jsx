import React, { Component } from 'react';
import {connect} from "react-redux";
import login from "../actions/loginActions";
import {errorAlert} from "../actions/signUpActions";

class Login extends Component {
    state = {  
        username: "",
        password: ""
    }

    loginUser = event => {
        event.preventDefault();
        const {username, password} = this.state;
        if (username || password) {
            this.props.loginUser({
                username,
                password
            });
        };
      };
    

    onChangeField = (event, fieldType) => {
        event.preventDefault();
        this.setFieldToState(event, fieldType);
    }

    setFieldToState = (event, fieldType) => {
        fieldType === "username" ? this.setState({
            username: event.target.value }) : fieldType === "password" && this.setState({
                password: event.target.value })
    };

    render() { 
        return ( 
            <React.Fragment>
                <div id="signup" className="container">
                <form className="form-horizontal" >
                    <fieldset>
                        <legend>SIgn Up</legend>
                        <div className="form-group">
                            <label htmlFor="inputUsername" className="col-lg-2 control-label">Username</label>
                            <div className="col-lg-10">
                                <input type="text"
                                    required={true}
                                    onChange={()=>(this.onChangeField(event, "username"))}
                                    className="form-control" id="inputUsername" placeholder="Username"  />
                            </div>
                        </div>

                        <div className="form-group">
                            <label htmlFor="inputPassword" className="col-lg-2 control-label">Password</label>
                            <div className="col-lg-10">
                                <input type="password" 
                                    required={true}
                                    onChange={()=>(this.onChangeField(event, "password"))}
                                    className="form-control" id="inputRepeatPassword" placeholder="Password" />
                            <div className="checkbox">
                            <label>
                                <input type="checkbox" /> Remember me
                            </label>
                            </div>
                        </div>
                        </div>
                        <div className="form-group">
                            <div className="col-lg-10 col-lg-offset-2">
                                <button type="reset" className="btn btn-default">Cancel</button>
                                <button 
                                    onClick={this.loginUser}
                                    type="submit" className="btn btn-primary">Submit</button>
                            </div>
                        </div>
                        
                    </fieldset>
                </form>
          
                </div>
                
            </React.Fragment>
         );
    };
};

export const mapStateToProps = state => state;

export const mapDispatchToProps = dispatch => ({
    loginUser: loginData => dispatch(login(loginData))
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);