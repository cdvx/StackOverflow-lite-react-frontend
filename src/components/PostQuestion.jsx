import React, { Component } from 'react';
import {connect} from "react-redux";
import askQuestion from "../actions/postQuestionActions";


// const QuestionModal = ({children}) => {
//     return ( 
//         <div class="modal">
//         <div class="modal-dialog">
//             <div class="modal-content">
//             <div class="modal-header">
//                 <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
//                 <h4 class="modal-title">Ask Question</h4>
//             </div>
//             <div class="modal-body">
//                 {children}
//             </div>
//             <div class="modal-footer">
//                 <button type="button" class="btn btn-default" data-dismiss="modal">Close</button>
//                 <button type="button" class="btn btn-primary">Save changes</button>
//             </div>
//             </div>
//         </div>
//         </div>
//      );
// }


class PostQuestion extends Component {
    state = {  
        topic: "",
        body: ""
    }

    postQuestion = event => {
        event.preventDefault();
        const {topic, body} = this.state;
        console.log(".....>", topic, body);
        this.props.postQuestion({
            topic,
            body
        });
      };
    

    onChangeField = (event, fieldType) => {
        event.preventDefault();
        this.setFieldToState(event, fieldType);
    }

    setFieldToState = (event, fieldType) => {
        fieldType === "topic" ? this.setState({
            topic: event.target.value }) : fieldType === "body" && this.setState({
                body: event.target.value })
    };

    render() { 
        return ( 
            <React.Fragment>
                
                <div id="signup" className="container">
                <form className="form-horizontal" >
                    <fieldset>


                        <legend>Post Question</legend>
                        <div className="form-group">
                            <label htmlFor="inputTopic" className="col-lg-2 control-label">Topic</label>
                            <div className="col-lg-10">
                                <input type="text" required onChange={()=>(this.onChangeField(event, "topic"))}
                                    className="form-control" id="inputTopic" placeholder="Topic"  />
                            </div>
                        </div>

                        {/* <div className="form-group">
                            <label htmlFor="inputPassword" className="col-lg-2 control-label">Password</label>
                            <div className="col-lg-10">
                                <input type="password" 
                                    required={true}
                                    onChange={()=>(this.onChangeField(event, "password"))}
                                    className="form-control" id="inputRepeatPassword" placeholder="Password" />
                            <div className="checkbox">
                            <label>
                                <input type="checkbox" /> Remember me
                            </label>
                            </div>
                        </div>*/}
                        {/* </div>  */}
                        <div className="form-group">
                        <label htmlFor="textArea" class="col-lg-2 control-label">Body</label>
                        <div className="col-lg-10">
                            <textarea 
                                onChange={()=>(this.onChangeField(event, "body"))}
                                className="form-control" rows="3" id="textArea"></textarea>
                            <span className="help-block">The body can be as long as you please.</span>
                        </div>
                        <div className="form-group">
                            <div className="col-lg-10 col-lg-offset-2">
                                {/* <button type="reset" className="btn btn-default">Cancel</button> */}
                                <button 
                                    onClick={this.postQuestion}
                                    type="submit" className="btn btn-primary mt-4">Post Question</button>
                            </div>
                        </div>
                        </div>
                    </fieldset>
                </form>
          
                </div>
                
            </React.Fragment>
         );
    };
};

export const mapStateToProps = state => state;

export const mapDispatchToProps = dispatch => ({
    postQuestion: questionData => dispatch(askQuestion(questionData))
});

export default connect(mapStateToProps, mapDispatchToProps)(PostQuestion);