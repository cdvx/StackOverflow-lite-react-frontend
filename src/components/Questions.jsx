import React, { Component } from 'react';
import { connect } from "react-redux";
import questions from "../actions/questionsAction";

export const Question = ({question}) => {
    return (  
        <React.Fragment >
            <blockquote id="blockQ">
                <p className="text-black" ><strong>Topic: </strong>{question.topic}</p>
                <p id="text-body1">{question.body}</p>
                <small  className="text-black"><strong>Author: </strong>{question.author}</small>
            </blockquote>
        </React.Fragment>
    );
}


const Info = ({messages}) => {
    return (  
        <React.Fragment>
            <tr className="info">
            <td>message</td>
            <td>{messages}</td>
            </tr>
        </React.Fragment>
    );
}

class Questions extends Component {
    state = {  
        questions: [],
        messages: ""
    };

    componentWillMount(){
        const {getQuestions} = this.props;
        getQuestions();
    }

    handleQuestionClick =  questionId => {               
        localStorage.setItem("questionId", questionId);
    }

    renderQuestions = questions => (questions.map(question=>{
        return (
        <React.Fragment>
                <tr >
                <td id="quest"><a id="select-question" href="/question" onClick={()=>(this.handleQuestionClick(question.questionId))}><Question question={question} /></a></td>
                </tr>
                <div id="space-this"></div>

        </React.Fragment>
        )
    }));
    render() { 
        const {results, messages} = this.props;
        console.log("props here>>", this.props, results, messages)
        return ( 
            <React.Fragment>
                <table className="table table-striped table-hover ">
                    <thead>
                        <tr>
                        <div className="container"><th><big >Questions</big></th></div>
                        </tr>
                    </thead>
                    <tbody>
                        {results && this.renderQuestions(results)}
                        {messages && <Info messages={messages}/>}
                    
                    </tbody>
                </table> 
            </React.Fragment>
         );
    }
}
 

export const mapStateToProps = ({questions}) => ({
    results: questions.questions,
    messages: questions.messages
});

export const mapDispatchToProps = dispatch => ({
    getQuestions: () => dispatch(questions()),
    getQuestion: questionId => dispatch(question(questionId))
});

export default connect(mapStateToProps, mapDispatchToProps)(Questions);