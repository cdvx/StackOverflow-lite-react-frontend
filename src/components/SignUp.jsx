import React, { Component } from 'react';
import {connect} from "react-redux";
import signup from '../actions/signUpActions';


const  Inputs = ({htmlFor, innerHtml, onChangeField, type1, type, placeholder}) => {
    return (  
        <React.Fragment>
            <div className="form-group">
                <label htmlFor={htmlFor} className="col-lg-2 control-label">{innerHtml}</label>
                <div className="col-lg-10">
                    <input type="text"
                        id={htmlFor}
                        onChange={(event)=>(onChangeField(event, type1))}
                        type={type}
                        className="form-control" placeholder={placeholder}  required/>
                </div>
            </div>
        </React.Fragment>
    );
};


class SignUp extends Component {
    state = {  
        username: "",
        email: "",
        password: "",
        repeatPassword: ""
    }

    signupUser = event => {
        event.preventDefault();
        const {username, password, email, repeatPassword} = this.state;
    
        if (username && email && password && repeatPassword) {
            this.props.signupUser({
                username,
                email,
                password,
                repeat_password: repeatPassword
            });
        };
      };
    

    onChangeField = (event, fieldType) => {
        event.preventDefault();
        this.setFieldToState(event, fieldType);
    }

    setFieldToState = (event, fieldType) => {
        if (fieldType === "username"){
            this.setState({
                username: event.target.value
            })
        }
        else if(fieldType === "email"){
            this.setState({
                email: event.target.value
            })
        }
        else {
            fieldType === "password" ? this.setState({
                password: event.target.value }) : fieldType === "repeatPassword" && this.setState({
                    repeatPassword: event.target.value })
        };
    };

    render() { 
        return ( 
            <React.Fragment>
                <div id="signup" className="container">
                <form className="form-horizontal" >
                    <fieldset>
                        <legend className="text-black">SIgn Up</legend>
                        
                        <Inputs
                            htmlFor={"inputUsername"}
                            innerHtml={"Username"}
                            onChangeField={this.onChangeField}
                            type1={"username"}
                            type={"username"}
                            placeholder={"Username"}
                        />

                        <Inputs
                            htmlFor={"inputEmail"}
                            innerHtml={"Email"}
                            onChangeField={this.onChangeField}
                            type1={"email"}
                            type={"email"}
                            placeholder={"Email"}
                        />

                        <Inputs
                            htmlFor={"inputPassword"}
                            innerHtml={"Password"}
                            onChangeField={this.onChangeField}
                            type1={"password"}
                            type={"password"}
                            placeholder={"Password"}
                        />

                        <Inputs
                            htmlFor={"inputRepeatPassword"}
                            innerHtml={"Repeat Password"}
                            onChangeField={this.onChangeField}
                            type1={"repeatPassword"}
                            type={"password"}
                            placeholder={"Repeat Password"}
                        />

                        <div className="form-group">
                            <div className="col-lg-10 col-lg-offset-2">
                                <button
                                    id="button-dark-blue" 
                                    onClick={this.signupUser}
                                    type="submit" className="btn btn-primary">Submit</button>
                            </div>
                        </div>
                        <div className="form-group">
                            <p>Already have an account?
                                <a href="/login">Login</a>
                            </p>
                        </div>
                    </fieldset>
                </form>
          
                </div>
                
            </React.Fragment>
         );
    };
};

export const mapStateToProps = state => state;

export const mapDispatchToProps = dispatch => ({
  signupUser: signupData => dispatch(signup(signupData))
});

export default connect(mapStateToProps, mapDispatchToProps)(SignUp);