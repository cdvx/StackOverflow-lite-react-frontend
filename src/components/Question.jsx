import React, { Component } from 'react';
import { connect } from 'react-redux';
import {Question} from "./Questions";
import question from "../actions/questionAction";

class ShowQuestion extends Component{
    state = {
        question: {}
    };

    componentWillMount(){
        const {getQuestion} = this.props;
        const questionId = localStorage.getItem("questionId");
        getQuestion(questionId);
    }

    showLoader = () => (
        <div className="text-center">
            <iframe src="https://giphy.com/embed/IwSG1QKOwDjQk" id="loadFrame" frameBorder="0" class="giphy-embed" >
            </iframe>
        </div>
    );

    render(){
        
        const {question, loading} = this.props.questionReducer;

        return (
            <React.Fragment>
                {loading === true ? this.showLoader() :
                <table className="table table-striped table-hover ">
                    <thead>
                        <tr>

                        <div className="container"><th><big>Question</big></th></div>

                        </tr>
                    </thead>
                    <tbody >
                        <tr >
                        <td id="quest"><Question question={question} /></td>

                        </tr>
                    </tbody>
                </table>}
            </React.Fragment>
        );
    }
};

export const mapStateToProps = ({questionReducer}) => ({
    questionReducer
});

export const mapDispatchToProps = dispatch => ({
    getQuestion: questionId => dispatch(question(questionId))
});

export default connect(mapStateToProps, mapDispatchToProps)(ShowQuestion);