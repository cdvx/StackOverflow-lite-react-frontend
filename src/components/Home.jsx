import React from 'react';
import Header from "./Header";
import Footer from "./Footer";
import SignUp from "./SignUp";
import "../assets/styles.css";

class Home extends React.Component {
  state = {
  };
  render() {
    return (
      <React.Fragment>
        <div className="jumbotron2">
        </div>
      </React.Fragment>
    );
  }
}

export default Home;
