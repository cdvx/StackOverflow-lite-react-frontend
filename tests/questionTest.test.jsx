import { mount, shallow } from "enzyme";

import React from "react";
import expect from "expect";
import { MemoryRouter } from "react-router-dom";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import { Question} from "../src/components/Questions";
import ShowQuestion, {mapStateToProps, mapDispatchToProps} from "../src/components/Question";


const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({ questionReducer: {question: {topic:"hfbcn", body:"fsdsds", author:"hcxhffd"}}, 
  getQuestions: jest.fn(),
  question: {topic:"hfbcn", body:"fsdsds", author:"hcxhffd"},
});

it("it should render component", () => {
  shallow(
    <MemoryRouter>
      <Provider store={store} question={props.question}><ShowQuestion /></Provider>
    </MemoryRouter>,
  );
});

const props = {
  question: {
    topic: "User with this name or author not found !",
    body:"ggggggg",
    author: "rfghhj"
  },
  messages: "ffggggg",
  onClick: jest.fn(),
  loginUser: jest.fn(),
  followedUsername: "cdvx",
};

describe("<ShowQuestion />", () => {
  test("renders the component", () => {
    const wrapper = shallow(<Provider store={store}><ShowQuestion /></Provider>);
    expect(wrapper).toMatchSnapshot();
  });
});


  describe("<Question />", () => {
    test("renders the component", () => {
      const wrapper = mount(<Question question={props.question}/>);
      expect(wrapper).toMatchSnapshot();
    });
  });



describe("test the login container", () => {
  const fakeEvent = { preventDefault: () => {}};
  localStorage.setItem("questionId", "234345")
  const wrapper = shallow(<Provider store={store}><ShowQuestion /></Provider>);
  wrapper.setState({
    queston: {topic:"hfbcn", body:"fsdsds", author:"hcxhffd"}
  });
  wrapper.setProps({getQuestion: jest.fn()});
  it("should mount the login component", () => {
    expect(wrapper.find("big")).toBeTruthy();
    expect(wrapper.find("Question")).toBeTruthy();
    expect(wrapper.find("table")).toBeTruthy();
    expect(wrapper.find("td")).toBeTruthy();

  });
});

describe("Login", () => {
  localStorage.setItem("questionId", "123");
  it("should show previously previously passed state", () => {
    const state = {
        "questionReducer": {question: {topic:"hfbcn", body:"fsdsds", author:"hcxhffd"}}
    };
    expect(mapStateToProps(state)).toEqual({
        question: undefined
      });

  });

  it("Should pass action to props on submit", () => {
    const dispatch = jest.fn();
    const fakeEvent = { preventDefault: () => {}};
    mapDispatchToProps(dispatch).getQuestion("123");
    expect((dispatch.mock.calls[0][0])).toBeTruthy();
  });
});


describe("Checks whether mapstate to props returns", () => {
  const expectedProp = {
    question: {topic:"hfbcn", body:"fsdsds", author:"hcxhffd"},
    getQuestion: jest.fn()
  };

  expect(mapStateToProps(expectedProp)).toBeTruthy();
});


describe("Checks whether mapdispatch to props returns", () => {
  const dispatch = jest.fn();
  mapDispatchToProps(dispatch).getQuestion("1234");
  expect(dispatch.mock.calls.length).toBe(1);
});
