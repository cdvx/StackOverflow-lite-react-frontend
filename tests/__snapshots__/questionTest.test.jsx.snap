// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Question /> renders the component 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <Question
    question={
      Object {
        "author": "rfghhj",
        "body": "ggggggg",
        "topic": "User with this name or author not found !",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "question": Object {
        "author": "rfghhj",
        "body": "ggggggg",
        "topic": "User with this name or author not found !",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": <blockquote
        id="blockQ"
      >
        <p
          class="text-black"
        >
          <strong>
            Topic: 
          </strong>
          User with this name or author not found !
        </p>
        <p
          id="text-body1"
        >
          ggggggg
        </p>
        <small
          class="text-black"
        >
          <strong>
            Author: 
          </strong>
          rfghhj
        </small>
      </blockquote>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="text-black"
          >
            <strong>
              Topic: 
            </strong>
            User with this name or author not found !
          </p>,
          <p
            id="text-body1"
          >
            ggggggg
          </p>,
          <small
            className="text-black"
          >
            <strong>
              Author: 
            </strong>
            rfghhj
          </small>,
        ],
        "id": "blockQ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <p
            class="text-black"
          >
            <strong>
              Topic: 
            </strong>
            User with this name or author not found !
          </p>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <strong>
                Topic: 
              </strong>,
              "User with this name or author not found !",
            ],
            "className": "text-black",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <strong>
                Topic: 
              </strong>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Topic: ",
              },
              "ref": null,
              "rendered": Array [
                "Topic: ",
              ],
              "type": "strong",
            },
            "User with this name or author not found !",
          ],
          "type": "p",
        },
        Object {
          "instance": <p
            id="text-body1"
          >
            ggggggg
          </p>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ggggggg",
            "id": "text-body1",
          },
          "ref": null,
          "rendered": Array [
            "ggggggg",
          ],
          "type": "p",
        },
        Object {
          "instance": <small
            class="text-black"
          >
            <strong>
              Author: 
            </strong>
            rfghhj
          </small>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <strong>
                Author: 
              </strong>,
              "rfghhj",
            ],
            "className": "text-black",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <strong>
                Author: 
              </strong>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Author: ",
              },
              "ref": null,
              "rendered": Array [
                "Author: ",
              ],
              "type": "strong",
            },
            "rfghhj",
          ],
          "type": "small",
        },
      ],
      "type": "blockquote",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "question": Object {
          "author": "rfghhj",
          "body": "ggggggg",
          "topic": "User with this name or author not found !",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": <blockquote
          id="blockQ"
        >
          <p
            class="text-black"
          >
            <strong>
              Topic: 
            </strong>
            User with this name or author not found !
          </p>
          <p
            id="text-body1"
          >
            ggggggg
          </p>
          <small
            class="text-black"
          >
            <strong>
              Author: 
            </strong>
            rfghhj
          </small>
        </blockquote>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="text-black"
            >
              <strong>
                Topic: 
              </strong>
              User with this name or author not found !
            </p>,
            <p
              id="text-body1"
            >
              ggggggg
            </p>,
            <small
              className="text-black"
            >
              <strong>
                Author: 
              </strong>
              rfghhj
            </small>,
          ],
          "id": "blockQ",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <p
              class="text-black"
            >
              <strong>
                Topic: 
              </strong>
              User with this name or author not found !
            </p>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <strong>
                  Topic: 
                </strong>,
                "User with this name or author not found !",
              ],
              "className": "text-black",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <strong>
                  Topic: 
                </strong>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Topic: ",
                },
                "ref": null,
                "rendered": Array [
                  "Topic: ",
                ],
                "type": "strong",
              },
              "User with this name or author not found !",
            ],
            "type": "p",
          },
          Object {
            "instance": <p
              id="text-body1"
            >
              ggggggg
            </p>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ggggggg",
              "id": "text-body1",
            },
            "ref": null,
            "rendered": Array [
              "ggggggg",
            ],
            "type": "p",
          },
          Object {
            "instance": <small
              class="text-black"
            >
              <strong>
                Author: 
              </strong>
              rfghhj
            </small>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <strong>
                  Author: 
                </strong>,
                "rfghhj",
              ],
              "className": "text-black",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <strong>
                  Author: 
                </strong>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Author: ",
                },
                "ref": null,
                "rendered": Array [
                  "Author: ",
                ],
                "type": "strong",
              },
              "rfghhj",
            ],
            "type": "small",
          },
        ],
        "type": "blockquote",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<ShowQuestion /> renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Connect(ShowQuestion) />
  </Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Connect(ShowQuestion) />,
      "value": Object {
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeState": Object {
          "getQuestions": [MockFunction],
          "question": Object {
            "author": "hcxhffd",
            "body": "fsdsds",
            "topic": "hfbcn",
          },
          "questionReducer": Object {
            "question": Object {
              "author": "hcxhffd",
              "body": "fsdsds",
              "topic": "hfbcn",
            },
          },
        },
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Connect(ShowQuestion) />,
        "value": Object {
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "storeState": Object {
            "getQuestions": [MockFunction],
            "question": Object {
              "author": "hcxhffd",
              "body": "fsdsds",
              "topic": "hfbcn",
            },
            "questionReducer": Object {
              "question": Object {
                "author": "hcxhffd",
                "body": "fsdsds",
                "topic": "hfbcn",
              },
            },
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
